src/bitkmer.rs

```diff
@@ -2,264 +2,20 @@
 
-static NUC2BIT_LOOKUP: [Option<u8>; 256] = [
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    Some(0),
-    None,
-    Some(1),
-    None,
-    None,
-    None,
-    Some(2),
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    Some(3),
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    Some(0),
-    None,
-    Some(1),
-    None,
-    None,
-    None,
-    Some(2),
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    Some(3),
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-    None,
-];
+const NUC2BIT_LOOKUP: [Option<u8>; 256] = {
+    let mut lookup = [None; 256];
```

Comment:
Maybe the `Option<u8>` is avoidable if you use a sentinel value instead (`255` would work), and then changing the `nuc2bti_lookup` function to return `None` if you find a `255`, or `Some(n)` if n is from a valid nucleotide

Reply:
Here I use `Option` because it allows to indicate when the value passed to the table is not A, C, T, G, a, c, t or g. What the tests that this lookup table replaces, did, I just maintain the same functionality as before.

But if it's up to me, values other than validated nucleotides would be transformed in a valid nucleotide arbitrary, which I do in  [my code](https://github.com/natir/cocktail/blob/master/src/kmer/mod.rs#L50).

Reply:
A more complet benchmark on different method to convert nucleotide in 2 bit representation is avaible in this [repo](https://github.com/natir/fbio)

Reply:
If you don't validate all the nucleotides prior to doing the bit-magic then you'll coerce all the N's into G's which introduces a new set of problems. When I ported this code over from Nim a long, long time ago I played around with that, but I had to add a guard clause to prevent bad nucleotides from creeping through which ended up being slightly slower than the match clause.

This new table-logic is much, much nicer though. :)

Reply:
If the user hasn't cleaned up his data before running my tools, that's his problem, garbage in garbage out.

More seriously I totally understand your point @bovee.